src/main/java/seedu/address/logic/commands/DeleteClaimCommand.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CLAIM_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INSURANCE_ID;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_CLIENTS;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.client.Client;
+import seedu.address.model.client.exceptions.ClaimException;
+import seedu.address.model.client.exceptions.InsurancePlanException;
+import seedu.address.model.client.insurance.InsurancePlan;
+import seedu.address.model.client.insurance.InsurancePlanFactory;
+import seedu.address.model.client.insurance.InsurancePlansManager;
+import seedu.address.model.client.insurance.claim.Claim;
+
+/**
+ * Removes a claim from an existing client with existing Insurance Plan in the address book.
+ */
+public class DeleteClaimCommand extends Command {
+    public static final String COMMAND_WORD = "deleteClaim";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes a claim from the insurance plan of the client identified "
+            + "by their client id. \n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + " INSURANCE_PLAN_ID (must be a valid ID), "
+            + " Claim_ID (must be a valid ID) \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_INSURANCE_ID + " 0 "
+            + PREFIX_CLAIM_ID + " B1234";
+
+    public static final String MESSAGE_DELETE_CLAIM_SUCCESS =
+            "Claim deleted from Client: %1$s, under Insurance plan %2$s, with Claim ID: %3$s";
+
+    public final Index index;
+    private final int insuranceId;
+    private final String claimID;
```

Comment:
Possibly could keep camelCase here with

`claimId`

Instead of 

`claimID`?

Reply:
Good catch, will make the changes